go-controller/pkg/clustermanager/default_network_cluster_controller.go

```diff
@@ -0,0 +1,400 @@
+
+	kube := &kube.Kube{
+		KClient: ovnClient.KubeClient,
+	}
+
+	stopChan := make(chan struct{})
+	wg := &sync.WaitGroup{}
+
+	var hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+	if config.HybridOverlay.Enabled {
+		hybridOverlaySubnetAllocator = subnetallocator.NewHostSubnetAllocator()
+	}
+	ncm := &defaultNetworkClusterController{
+		networkClusterControllerBase: networkClusterControllerBase{
+			kube:         kube,
+			watchFactory: wf,
+			stopChan:     stopChan,
+			wg:           wg,
+		},
+		defaultNetworkController:     newNodeNetworkController(kube, wf, ovntypes.DefaultNetworkName),
+		hybridOverlaySubnetAllocator: hybridOverlaySubnetAllocator,
+	}
+
+	ncm.initRetryFramework()
+	return ncm
+}
+
+func (ncm *defaultNetworkClusterController) initRetryFramework() {
+	ncm.retryNodes = ncm.newRetryFramework(factory.NodeType)
+}
+
+// Init initializes the default network subnet allocator ranges
+// and hybrid network subnet allocator ranges if hybrid overlay is enabled.
+func (ncm *defaultNetworkClusterController) Init() error {
+	if err := ncm.defaultNetworkController.InitSubnetAllocatorRanges(config.Default.ClusterSubnets); err != nil {
+		return fmt.Errorf("failed to initialize host subnet allocator ranges: %w", err)
+	}
+
+	if config.HybridOverlay.Enabled {
+		if err := ncm.hybridOverlaySubnetAllocator.InitRanges(config.HybridOverlay.ClusterSubnets); err != nil {
+			return fmt.Errorf("failed to initialize hybrid overlay subnet allocator ranges: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (ncm *defaultNetworkClusterController) Stop() {
+	close(ncm.stopChan)
+	ncm.wg.Wait()
+}
+
+// Run starts watching kubernetes nodes.
+func (ncm *defaultNetworkClusterController) Run() error {
+	return ncm.watchNodes()
+}
+
+// WatchNodes starts the watching of node resource and calls
+// back the appropriate handler logic
+func (ncm *defaultNetworkClusterController) watchNodes() error {
+	if ncm.nodeHandler != nil {
+		return nil
+	}
+
+	handler, err := ncm.retryNodes.WatchResource()
+	if err != nil {
+		return err
+	}
+
+	ncm.nodeHandler = handler
+	return nil
+}
+
+// defaultNetworkClusterControllerEventHandler object handles the events
+// from retry framework.
+type defaultNetworkClusterControllerEventHandler struct {
+	objretry.EventHandler
+
+	objType  reflect.Type
+	ncm      *defaultNetworkClusterController
+	syncFunc func([]interface{}) error
+}
+
+func (ncm *defaultNetworkClusterController) newRetryFramework(objectType reflect.Type) *objretry.RetryFramework {
+	resourceHandler := &objretry.ResourceHandler{
+		HasUpdateFunc:          hasResourceAnUpdateFunc(objectType),
+		NeedsUpdateDuringRetry: false,
+		ObjType:                objectType,
+		EventHandler: &defaultNetworkClusterControllerEventHandler{
+			objType:  objectType,
+			ncm:      ncm,
+			syncFunc: nil,
+		},
+	}
+	return objretry.NewRetryFramework(ncm.stopChan, ncm.wg, ncm.watchFactory, resourceHandler)
```

Comment:
IMO, I think all of this should be moved to Run because it requires stop chan and wait group. No need to store them in the type struct then. WDYT?